Device Control Panel shows entities but loads endlessly

<!-- READ THIS FIRST:
- Make sure you run the latest version of the Android app
- Make sure you run the latest version of Home Assistant
- Make sure to check the Companion docs for troubleshooting and configuration: https://companion.home-assistant.io/
- Make sure the bug you found is not already reported, we love to put work in bugfixes instead of closing duplicate bug reports
  DO NOT DELETE ANY TEXT from this template! All requested information is important.
-->

**Home Assistant Android version:**
beta-1916-8a8772d5-full

**Android version:**
12

**Phone model:**
Oppo Find X3 Pro

**Home Assistant version:**
Home Assistant 2021.12.10

**Last working Home Assistant release (if known):**
not known

**Description of problem:**
For a few days, if not weeks, the Android menu for Device Control has not been working properly. The problem already existed under Android 11, is still present after the update to Android 12.

The Home Assistant Android app itself works perfectly. However, the entities in the Device Control menu are displayed, but they always load. See screenshot for explanation. 

Other apps that can be used via Device Control (e.g. Hue Essentials) work fine. 

I have already completely deleted and reinstalled the app. Unfortunately, this did not help either. 


**Traceback **

```
--------- beginning of main
01-20 18:31:34.149 22380 22380 E mpanion.androi: Unknown bits set in runtime_flags: 0x40000000
01-20 18:31:34.152 22380 22380 E mpanion.androi: Not starting debugger since process cannot load the jdwp agent.
01-20 18:31:34.153 22380 22380 D ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
01-20 18:31:34.155 22380 22380 I OneTrace: Mark active for pid=22380? true
01-20 18:31:34.165 22380 22380 D CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10280; state: ENABLED
01-20 18:31:34.165 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusAutoResolutionFeature
01-20 18:31:34.165 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: getOplusAutoResolutionFeature
01-20 18:31:34.169 22380 22392 D ColorModeChangeItem: preExecute mColorMode=0,token=android.os.BinderProxy@ef7133b
01-20 18:31:34.172 22380 22380 W mpanion.androi: Entry not found
01-20 18:31:34.172 22380 22380 W mpanion.androi: Entry not found
01-20 18:31:34.173 22380 22380 W mpanion.androi: Entry not found
01-20 18:31:34.173 22380 22380 W mpanion.androi: Entry not found
01-20 18:31:34.175 22380 22380 W mpanion.androi: JIT profile information will not be recorded: profile file does not exist.
01-20 18:31:34.175 22380 22380 W mpanion.androi: JIT profile information will not be recorded: profile file does not exist.
01-20 18:31:34.175 22380 22380 W mpanion.androi: JIT profile information will not be recorded: profile file does not exist.
01-20 18:31:34.175 22380 22380 W mpanion.androi: JIT profile information will not be recorded: profile file does not exist.
--------- beginning of system
01-20 18:31:34.176 22380 22380 I Quality : LaunchTime2.0: io.homeassistant.companion.android createOrUpdateClassLoaderLocked took time: 6
01-20 18:31:34.180 22380 22380 V GraphicsEnvironment: ANGLE Developer option for 'io.homeassistant.companion.android' set to: 'default'
01-20 18:31:34.180 22380 22380 V GraphicsEnvironment: App is not on the allowlist for updatable production driver.
01-20 18:31:34.181 22380 22380 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
01-20 18:31:34.181 22380 22380 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
01-20 18:31:34.198 22380 22380 I FirebaseApp: Device unlocked: initializing all Firebase APIs for app [DEFAULT]
01-20 18:31:34.206 22380 22380 I FirebaseInitProvider: FirebaseApp initialization successful
01-20 18:31:34.207 22380 22380 D WM-WrkMgrInitializer: Initializing WorkManager with default configuration.
01-20 18:31:34.217 22380 22380 D Sentry  : io.sentry.auto-init read: false
01-20 18:31:34.217 22380 22380 I Sentry  : Retrieving auto-init from AndroidManifest.xml
01-20 18:31:34.217 22380 22405 I DynamiteModule: Considering local module com.google.android.gms.measurement.dynamite:55 and remote module com.google.android.gms.measurement.dynamite:59
01-20 18:31:34.217 22380 22405 I DynamiteModule: Selected remote version of com.google.android.gms.measurement.dynamite, version >= 59
01-20 18:31:34.218 22380 22405 V DynamiteModule: Dynamite loader version >= 2, using loadModule2NoCrashUtils
01-20 18:31:34.229 22380 22406 D CompatibilityChangeReporter: Compat change id reported: 160794467; UID 10280; state: ENABLED
01-20 18:31:34.238 22380 22380 I Quality : BindApplication: io.homeassistant.companion.android
01-20 18:31:34.243 22380 22417 E mpanion.androi: Loaded layer handle (14027242074726323083) for layer /my_product/lib64/libcolorx-loader.so
01-20 18:31:34.243 22380 22417 E mpanion.androi: Looking for entrypoint ColorX_Check
01-20 18:31:34.243 22380 22417 E mpanion.androi: Found ColorX_Check for /my_product/lib64/libcolorx-loader.so
01-20 18:31:34.243 22380 22417 E mpanion.androi: Check failed
01-20 18:31:34.263 22380 22418 I FA      : App measurement initialized, version: 50018
01-20 18:31:34.263 22380 22418 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
01-20 18:31:34.263 22380 22418 I FA      : To enable faster debug mode event logging run:
01-20 18:31:34.263 22380 22418 I FA      :   adb shell setprop debug.firebase.analytics.app io.homeassistant.companion.android
01-20 18:31:34.332 22380 22380 I WebViewFactory: Loading com.google.android.webview version 97.0.4692.87 (code 469208734)
01-20 18:31:34.344 22380 22380 I cr_WVCFactoryProvider: Loaded version=97.0.4692.87 minSdkVersion=29 isBundle=true multiprocess=true packageId=2
01-20 18:31:34.349 22380 22380 I cr_LibraryLoader: Successfully loaded native library
01-20 18:31:34.350 22380 22380 I cr_CachingUmaRecorder: Flushed 8 samples from 8 histograms.
01-20 18:31:34.363 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: Unknow feature:IOplusTextViewRTLUtilForUG
01-20 18:31:34.411 22380 22380 D CompatibilityChangeReporter: Compat change id reported: 171228096; UID 10280; state: ENABLED
01-20 18:31:34.435 22380 22380 E OplusCustomizeRestrictionManager: sInstance is null, start a new sInstance
01-20 18:31:34.450 22380 22380 D RenderScript HIDL Adaptation: IRenderScriptDevice::getService()
01-20 18:31:34.450 22380 22380 D hw-ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
01-20 18:31:34.450 22380 22380 D RenderScript HIDL Adaptation: IRenderScriptDevice::getService() returned 0x0
01-20 18:31:34.450 22380 22380 D RenderScript HIDL Adaptation: Using Fallback Path.
01-20 18:31:34.455 22380 22380 D mpanion.androi: Successfully queried cache dir: /data/user_de/0/io.homeassistant.companion.android/code_cache
01-20 18:31:34.455 22380 22380 D RenderScript: Setting cache dir: /data/user_de/0/io.homeassistant.companion.android/code_cache
01-20 18:31:34.460 22380 22380 I System.out: recordForce value 5
01-20 18:31:34.460 22380 22380 D WebviewActivity: Current webview package com.google.android.webview and version 97.0.4692.87
01-20 18:31:34.464 22380 22418 I FA      : Tag Manager is not found and thus will not be used
01-20 18:31:34.475 22380 22380 D OplusInputMethodUtil: init DEBUG to false, update DEBUG_IME to false
01-20 18:31:34.475 22380 22380 D OplusInputMethodUtil: init DEBUG to false, update DEBUG_IME to false
01-20 18:31:34.479 22380 22380 I OplusFeatureCache: Milliseconds spent on init(): 1
01-20 18:31:34.479 22380 22380 D OplusSlideAnimOptHelper: io.homeassistant.companion.android scrollEffectOpt enable = false
01-20 18:31:34.483 22380 22415 D ExtensionsLoader: createInstance(64bit) : createExtendedFactory
01-20 18:31:34.483 22380 22415 E ExtensionsLoader: dlopen failed: library "libhwuiextimpl.so" not found
01-20 18:31:34.497 22380 22380 D LocBroadcastReceiver: Received location update.
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Last Location: 
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Coords:(XX, 7.XX)
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Accuracy: XX
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Bearing: XX
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Begin evaluating if location update should be skipped
01-20 18:31:34.501 22380 22380 D LocBroadcastReceiver: Received location that is 344 milliseconds old, 1642699894157 compared to 1642699894501 with source fused
01-20 18:31:34.502 22380 22380 I Choreographer: Skipped 33 frames!  The application may be doing too much work on its main thread.
01-20 18:31:34.502 22380 22380 I Quality : Skipped: false 33
01-20 18:31:34.507 22380 22410 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:34.519 22380 22380 V OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 3216 getDecorView.143247381
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: QUALCOMM build                   : 3b0ebcfc5c, I200588c14c
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Build Date                       : 10/27/21
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.35.01.03
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Local Branch                     : 
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Remote Branch                    : 
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Remote Branch                    : 
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Reconstruct Branch               : 
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch64
01-20 18:31:34.519 22380 22415 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
01-20 18:31:34.522 22380 22415 I AdrenoGLES-0: PFP: 0x016dc094, ME: 0x00000000
01-20 18:31:34.549 22380 22380 D LastUpdate: Last update is android.net.wifi.STATE_CHANGE
01-20 18:31:34.552 22380 22397 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
01-20 18:31:34.553 22380 22380 D LastUpdate: Last update is android.media.RINGER_MODE_CHANGED
01-20 18:31:34.556 22380 22412 D ActivitySM: Unregistering for activity updates.
01-20 18:31:34.557 22380 22412 D ActivitySM: Registering for activity updates.
01-20 18:31:34.558 22380 22411 D ActivitySM: Unregistering for activity updates.
01-20 18:31:34.558 22380 22411 D ActivitySM: Registering for activity updates.
01-20 18:31:34.559 22380 22411 D ActivitySM: Registering for sleep updates
01-20 18:31:34.559 22380 22412 D ActivitySM: Registering for sleep updates
01-20 18:31:34.560 22380 22380 I Quality : Skipped: false 5
01-20 18:31:34.561 22380 22411 D ActivitySM: Registering for both sleep confidence and segment updates
01-20 18:31:34.563 22380 22412 D ActivitySM: Registering for both sleep confidence and segment updates
01-20 18:31:34.564 22380 22496 D SensorWorker: Updating all Sensors.
01-20 18:31:34.564 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.565 22380 22414 I WM-Processor: Moving WorkSpec (4d3e482d-4562-40c4-8b7e-c943444fd068) to the foreground
01-20 18:31:34.572 22380 22490 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.572 22380 22440 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.573 22380 22380 I WM-SystemFgDispatcher: Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android/androidx.work.impl.foreground.SystemForegroundService (has extras) }
01-20 18:31:34.575 22380 22440 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.577 22380 22410 D LastUpdate: Last update is SensorWorker
01-20 18:31:34.584 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:34.591 22380 22410 D ActivitySM: Unregistering for activity updates.
01-20 18:31:34.591 22380 22410 D ActivitySM: Registering for activity updates.
01-20 18:31:34.592 22380 22410 D ActivitySM: Registering for sleep updates
01-20 18:31:34.593 22380 22410 D ActivitySM: Registering for both sleep confidence and segment updates
01-20 18:31:34.602 22380 22380 D ActivitySM: Successfully registered for sleep updates
01-20 18:31:34.607 22380 22380 D ActivitySM: Successfully registered for sleep updates
01-20 18:31:34.608 22380 22496 D LocBroadcastReceiver: Location update sent successfully
01-20 18:31:34.616 22380 22380 D ActivitySM: Received sleep update
01-20 18:31:34.617 22380 22380 D ActivitySM: Sleep segment event detected
01-20 18:31:34.617 22380 22380 D ActivitySM: Sleep segment has an actual event
01-20 18:31:34.625 22380 22380 D ActivitySM: Received sleep update
01-20 18:31:34.625 22380 22380 D ActivitySM: Sleep segment event detected
01-20 18:31:34.625 22380 22380 D ActivitySM: Sleep segment has an actual event
01-20 18:31:34.627 22380 22380 D ActivitySM: Successfully registered for sleep updates
01-20 18:31:34.636 22380 22380 D ActivitySM: Received sleep update
01-20 18:31:34.637 22380 22380 D ActivitySM: Sleep segment event detected
01-20 18:31:34.637 22380 22380 D ActivitySM: Sleep segment has an actual event
01-20 18:31:34.641 22380 22411 I AudioManager: In isSpeakerphoneOn(), calling application: io.homeassistant.companion.android
01-20 18:31:34.642 22380 22412 I AudioManager: In isSpeakerphoneOn(), calling application: io.homeassistant.companion.android
01-20 18:31:34.644 22380 22411 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=3, streamType=4
01-20 18:31:34.644 22380 22410 I AudioManager: In isSpeakerphoneOn(), calling application: io.homeassistant.companion.android
01-20 18:31:34.644 22380 22412 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=3, streamType=4
01-20 18:31:34.646 22380 22411 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=0
01-20 18:31:34.647 22380 22412 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=0
01-20 18:31:34.647 22380 22410 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=3, streamType=4
01-20 18:31:34.648 22380 22411 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=1, streamType=3
01-20 18:31:34.648 22380 22412 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=1, streamType=3
01-20 18:31:34.648 22380 22410 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=0
01-20 18:31:34.649 22380 22411 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=2
01-20 18:31:34.649 22380 22412 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=2
01-20 18:31:34.650 22380 22410 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=1, streamType=3
01-20 18:31:34.651 22380 22410 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=2
01-20 18:31:34.692 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:34.704 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:34.712 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.724 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.725 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.726 22380 22411 V SystemSensorManagerExtImpl: RegisterListener tcs3408 Ambient Light Sensor Non-wakeup type:5 delay:200000us by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:34.726 22380 22412 V SystemSensorManagerExtImpl: RegisterListener tcs3408 Ambient Light Sensor Non-wakeup type:5 delay:200000us by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:34.726 22380 22411 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.726 22380 22410 V SystemSensorManagerExtImpl: RegisterListener tcs3408 Ambient Light Sensor Non-wakeup type:5 delay:200000us by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:34.726 22380 22411 D LightSensor: Light sensor listener registered
01-20 18:31:34.726 22380 22410 D LightSensor: Light sensor listener registered
01-20 18:31:34.726 22380 22412 D LightSensor: Light sensor listener registered
01-20 18:31:34.727 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:34.728 22380 22496 D LocBroadcastReceiver: Registering for zone based location updates
01-20 18:31:34.728 22380 22380 D LightSensor: Light sensor listener unregistered
01-20 18:31:34.728 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.732 22380 22496 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:34.733 22380 22380 D LocBroadcastReceiver: Not requesting accurate location, last accurate location was too recent
01-20 18:31:34.737 22380 22380 D LocBroadcastReceiver: Not requesting accurate location, last accurate location was too recent
01-20 18:31:34.743 22380 22380 D LocBroadcastReceiver: Not requesting accurate location, last accurate location was too recent
01-20 18:31:34.743 22380 22497 D LocBroadcastReceiver: Registering for location updates.
01-20 18:31:34.745 22380 22495 D LocBroadcastReceiver: High accuracy mode parameters changed. Disable high accuracy mode.
01-20 18:31:34.750 22380 22495 D HighAccLocService: Try stopping high accuracy location service...
01-20 18:31:34.750 22380 22495 D LocBroadcastReceiver: Registering for location updates.
01-20 18:31:34.751 22380 22495 D LocBroadcastReceiver: High accuracy mode geo parameters changed. Reconfigure zones.
01-20 18:31:34.751 22380 22495 D LocBroadcastReceiver: Removing geofence location requests.
01-20 18:31:34.756 22380 22380 I WM-SystemFgDispatcher: Stopping foreground service
01-20 18:31:34.757 22380 22495 D LocBroadcastReceiver: Registering for zone based location updates
01-20 18:31:34.760 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:34.762 22380 22380 I WM-SystemFgDispatcher: Stopping foreground service
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: Work [ id=4d3e482d-4562-40c4-8b7e-c943444fd068, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ] was cancelled
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: java.util.concurrent.CancellationException: Task was cancelled.
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1184)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:514)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:311)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
01-20 18:31:34.785 22380 22484 I WM-WorkerWrapper: 	at java.lang.Thread.run(Thread.java:920)
01-20 18:31:34.789 22380 22380 I WM-SystemFgDispatcher: Stopping foreground service
01-20 18:31:34.792 22380 22411 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.792 22380 22497 D SensorWorker: Updating all Sensors.
01-20 18:31:34.794 22380 22410 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.794 22380 22412 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.795 22380 22406 I WM-Processor: Moving WorkSpec (beda43e8-7ff9-4b2d-b201-c3fcb867db41) to the foreground
01-20 18:31:34.800 22380 22380 I WM-SystemFgDispatcher: Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android/androidx.work.impl.foreground.SystemForegroundService (has extras) }
01-20 18:31:34.801 22380 22495 D LastUpdate: Last update is SensorWorker
01-20 18:31:34.808 22380 22495 D ActivitySM: Unregistering for activity updates.
01-20 18:31:34.808 22380 22495 D ActivitySM: Registering for activity updates.
01-20 18:31:34.810 22380 22411 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.812 22380 22412 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.813 22380 22410 D WifiManager: getScanResults check resulttrue
01-20 18:31:34.824 22380 22411 D NextAlarm: Next alarm is scheduled by net.dinglisch.android.taskerm with trigger time 1642713300000
01-20 18:31:34.825 22380 22412 D NextAlarm: Next alarm is scheduled by net.dinglisch.android.taskerm with trigger time 1642713300000
01-20 18:31:34.825 22380 22410 D NextAlarm: Next alarm is scheduled by net.dinglisch.android.taskerm with trigger time 1642713300000
01-20 18:31:34.854 22380 22495 I AudioManager: In isSpeakerphoneOn(), calling application: io.homeassistant.companion.android
01-20 18:31:34.856 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=3, streamType=4
01-20 18:31:34.859 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=0
01-20 18:31:34.863 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=1, streamType=3
01-20 18:31:34.867 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=2
01-20 18:31:34.899 22380 22410 V SystemSensorManagerExtImpl: RegisterListener stk33502 Proximity Sensor Wakeup type:8 delay:200000us by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:34.900 22380 22410 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.907 22380 22412 V SystemSensorManagerExtImpl: RegisterListener stk33502 Proximity Sensor Wakeup type:8 delay:200000us by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:34.908 22380 22411 V SystemSensorManagerExtImpl: RegisterListener stk33502 Proximity Sensor Wakeup type:8 delay:200000us by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:34.909 22380 22410 D ProximitySensor: Proximity sensor listener registered
01-20 18:31:34.909 22380 22412 D ProximitySensor: Proximity sensor listener registered
01-20 18:31:34.909 22380 22411 D ProximitySensor: Proximity sensor listener registered
01-20 18:31:34.913 22380 22410 V SystemSensorManagerExtImpl: RegisterListener pedometer  Non-wakeup type:19 delay:200000us by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:34.913 22380 22410 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.913 22380 22412 V SystemSensorManagerExtImpl: RegisterListener pedometer  Non-wakeup type:19 delay:200000us by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:34.914 22380 22380 V SystemSensorManagerExtImpl: step counter dispatchSensorEvent step 0.0 to + io.homeassistant.companion.android.sensors.StepsSensorManager@6f10403
01-20 18:31:34.914 22380 22410 D StepsSensor: Steps sensor listener registered
01-20 18:31:34.914 22380 22411 V SystemSensorManagerExtImpl: RegisterListener pedometer  Non-wakeup type:19 delay:200000us by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:34.914 22380 22411 D StepsSensor: Steps sensor listener registered
01-20 18:31:34.914 22380 22412 D StepsSensor: Steps sensor listener registered
01-20 18:31:34.915 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:34.916 22380 22380 D StepsSensor: Steps sensor listener unregistered
01-20 18:31:34.916 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.929 22380 22410 D StorageSensor: PATHS SD 1
01-20 18:31:34.930 22380 22410 D StorageSensor: PATH /storage/emulated/0/Android/data/io.homeassistant.companion.android/files is mounted true and removable is false
01-20 18:31:34.931 22380 22411 D StorageSensor: PATHS SD 1
01-20 18:31:34.932 22380 22412 D StorageSensor: PATHS SD 1
01-20 18:31:34.932 22380 22411 D StorageSensor: PATH /storage/emulated/0/Android/data/io.homeassistant.companion.android/files is mounted true and removable is false
01-20 18:31:34.933 22380 22412 D StorageSensor: PATH /storage/emulated/0/Android/data/io.homeassistant.companion.android/files is mounted true and removable is false
01-20 18:31:34.939 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:34.941 22380 22380 D ProximitySensor: Proximity sensor listener unregistered
01-20 18:31:34.941 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:34.976 22380 22410 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:34.978 22380 22411 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:34.986 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:34.994 22380 22412 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:35.009 22380 22380 D LocBroadcastReceiver: Received geofence update.
01-20 18:31:35.012 22380 22380 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:35.030 22380 22495 V SystemSensorManagerExtImpl: RegisterListener tcs3408 Ambient Light Sensor Non-wakeup type:5 delay:200000us by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:35.030 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.041 22380 22495 D LightSensor: Light sensor listener registered
01-20 18:31:35.071 22380 22495 D WifiManager: getScanResults check resulttrue
01-20 18:31:35.083 22380 22495 D WifiManager: getScanResults check resulttrue
01-20 18:31:35.094 22380 22495 D NextAlarm: Next alarm is scheduled by net.dinglisch.android.taskerm with trigger time XX
01-20 18:31:35.116 22380 22495 V SystemSensorManagerExtImpl: RegisterListener stk33502 Proximity Sensor Wakeup type:8 delay:200000us by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:35.116 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.123 22380 22495 D ProximitySensor: Proximity sensor listener registered
01-20 18:31:35.126 22380 22495 V SystemSensorManagerExtImpl: RegisterListener pedometer  Non-wakeup type:19 delay:200000us by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:35.126 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.127 22380 22495 D StepsSensor: Steps sensor listener registered
01-20 18:31:35.132 22380 22495 D StorageSensor: PATHS SD 1
01-20 18:31:35.133 22380 22495 D StorageSensor: PATH /storage/emulated/0/Android/data/io.homeassistant.companion.android/files is mounted true and removable is false
01-20 18:31:35.141 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:35.153 22380 22380 D LocBroadcastReceiver: Event sent to Home Assistant
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Last Location: 
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Coords:(47.3857959, 7.9452976)
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Accuracy: 4.137
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Bearing: 203.37372
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Begin evaluating if location update should be skipped
01-20 18:31:35.154 22380 22380 D LocBroadcastReceiver: Skipping old location update since time is before the last one we sent, received: 1642699894203 last sent: 1642699894501
01-20 18:31:35.155 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:35.156 22380 22380 D LightSensor: Light sensor listener unregistered
01-20 18:31:35.156 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.156 22380 22380 V SystemSensorManagerExtImpl: step counter dispatchSensorEvent step 0.0 to + io.homeassistant.companion.android.sensors.StepsSensorManager@6f10403
01-20 18:31:35.158 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:35.158 22380 22380 D StepsSensor: Steps sensor listener unregistered
01-20 18:31:35.158 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.159 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:35.161 22380 22380 D ProximitySensor: Proximity sensor listener unregistered
01-20 18:31:35.161 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:35.167 22380 22380 D LocBroadcastReceiver: Not requesting accurate location, last accurate location was too recent
01-20 18:31:35.277 22380 22410 E SensorReceiver: Exception while updating sensors.
01-20 18:31:35.277 22380 22410 E SensorReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: kotlinx.coroutines.JobCancellationException: Job was cancelled; job=JobImpl{Cancelling}@4972f9
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl.updateSensors(IntegrationRepositoryImpl.kt:598)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl$updateSensors$1.invokeSuspend(Unknown Source:15)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:39)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
01-20 18:31:35.277 22380 22410 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
01-20 18:31:35.277 22380 22410 E SensorReceiver: Caused by: kotlinx.coroutines.JobCancellationException: Job was cancelled; job=JobImpl{Cancelling}@4972f9
01-20 18:31:35.291 22380 22380 D ActivitySM: Received activity update.
01-20 18:31:36.203 22380 22484 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=beda43e8-7ff9-4b2d-b201-c3fcb867db41, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
01-20 18:31:36.208 22380 22380 I WM-SystemFgDispatcher: Stopping foreground service
01-20 18:31:37.103 22380 22380 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:37.103 22380 22380 D UrlRepository: Using internal URL
01-20 18:31:37.127 22380 22468 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:37.256 22380 22380 E WebviewActivity: onReceivedHttpError: android.webkit.WebResourceResponse@be6879b
01-20 18:31:37.283 22380 22380 I Quality : Skipped: true 1
01-20 18:31:37.290 22380 22380 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:37.290 22380 22380 D UrlRepository: Using internal URL
01-20 18:31:37.298 22380 22468 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:37.424 22380 22563 D WebviewActivity: External bus {"type":"connection-status","payload":{"event":"connected"},"id":1}
01-20 18:31:37.619 22380 22563 D WebviewActivity: External bus {"type":"theme-update","id":2}
01-20 18:31:37.626 22380 22380 D WebviewActivity: Color from webview is "rgb(23, 23, 23)-SPACER-rgb(23, 23, 23)"
01-20 18:31:37.626 22380 22380 D WebviewActivity: Color from webview is "rgb(23, 23, 23)-SPACER-rgb(23, 23, 23)"
01-20 18:31:37.626 22380 22495 D WebViewPresenterImpl: Try getting color from webview color "rgb(23, 23, 23)".
01-20 18:31:37.627 22380 22495 I WebViewPresenterImpl: Found color -15263977.
01-20 18:31:37.627 22380 22410 D WebViewPresenterImpl: Try getting color from webview color "rgb(23, 23, 23)".
01-20 18:31:37.627 22380 22410 I WebViewPresenterImpl: Found color -15263977.
01-20 18:31:37.660 22380 22563 D WebviewActivity: External bus {"type":"config/get","id":3}
01-20 18:31:37.661 22380 22380 D WebviewActivity: externalBus({"id":3,"type":"result","success":true,"result":{"hasSettingsScreen":true,"canWriteTag":true,"hasExoPlayer":true}});
01-20 18:31:37.700 22380 22380 D WebviewActivity: Callback null
01-20 18:31:37.700 22380 22380 D WebviewActivity: Color from webview is "rgb(23, 23, 23)-SPACER-rgb(23, 23, 23)"
01-20 18:31:37.700 22380 22380 D WebviewActivity: Color from webview is "rgb(23, 23, 23)-SPACER-rgb(23, 23, 23)"
01-20 18:31:37.700 22380 22495 D WebViewPresenterImpl: Try getting color from webview color "rgb(23, 23, 23)".
01-20 18:31:37.700 22380 22495 I WebViewPresenterImpl: Found color -15263977.
01-20 18:31:37.701 22380 22495 D WebViewPresenterImpl: Try getting color from webview color "rgb(23, 23, 23)".
01-20 18:31:37.701 22380 22495 I WebViewPresenterImpl: Found color -15263977.
01-20 18:31:38.144 22380 22380 I Quality : Skipped: true 4
01-20 18:31:38.393 22380 22380 I Quality : Skipped: false 1
01-20 18:31:38.422 22380 22380 D OplusSystemUINavigationGesture: [GESTURE_BUTTON] swipe from 0
01-20 18:31:38.422 22380 22380 D OplusSystemUINavigationGesture: [GESTURE_BUTTON] Hit Gesture Region !
01-20 18:31:38.621 22380 22380 I Quality : Skipped: true 1
01-20 18:31:38.746 22380 22380 I Quality : Skipped: false 2
01-20 18:31:38.855 22380 22380 I Quality : Skipped: true 1
01-20 18:31:40.057 22380 22380 D LocBroadcastReceiver: Received geofence update.
01-20 18:31:40.061 22380 22380 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:40.223 22380 22380 D LocBroadcastReceiver: Event sent to Home Assistant
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Last Location: 
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Coords:(XX, XX)
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Accuracy: XX
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Bearing: XX
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Begin evaluating if location update should be skipped
01-20 18:31:40.225 22380 22380 D LocBroadcastReceiver: Skipping old location update since time is before the last one we sent, received: 1642699894203 last sent: 1642699894501
01-20 18:31:40.227 22380 22380 I Quality : Skipped: true 19
01-20 18:31:40.296 22380 22380 I Quality : Skipped: false 2
01-20 18:31:41.445 22380 22563 D WebviewActivity: External bus {"type":"config_screen/show","id":4}
01-20 18:31:41.465 22380 22393 D ColorModeChangeItem: preExecute mColorMode=0,token=android.os.BinderProxy@6cb2f40
01-20 18:31:41.469 22380 22380 I Quality : Skipped: false 1
01-20 18:31:41.488 22380 22380 D UrlRepository: Using external URL
01-20 18:31:41.488 22380 22380 D UrlRepository: Using external URL
01-20 18:31:41.489 22380 22380 D UrlRepository: Using internal URL
01-20 18:31:41.489 22380 22380 D UrlRepository: Using internal URL
01-20 18:31:41.499 22380 22380 D ScrollOptimizationHelper: can't find the config file for optimizaiton
01-20 18:31:41.499 22380 22380 D ScrollOptimizationHelper: will not debug for debug is false
01-20 18:31:41.500 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:41.500 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:41.500 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:41.514 22380 22380 I Quality : Skipped: false 4
01-20 18:31:41.525 22380 22380 V OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 3216 getDecorView.220028422
01-20 18:31:41.575 22380 22393 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
01-20 18:31:41.577 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.589 22380 22380 I Quality : Skipped: false 8
01-20 18:31:41.604 22380 22380 I Quality : Skipped: false 1
01-20 18:31:41.604 22380 22410 D SensorWorker: Updating all Sensors.
01-20 18:31:41.605 22380 22484 I WM-Processor: Moving WorkSpec (341c3df7-5d2c-412c-b79f-429fc7d31d4d) to the foreground
01-20 18:31:41.620 22380 22495 D LastUpdate: Last update is SensorWorker
01-20 18:31:41.622 22380 22380 I WM-SystemFgDispatcher: Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android/androidx.work.impl.foreground.SystemForegroundService (has extras) }
01-20 18:31:41.622 22380 22495 D ActivitySM: Unregistering for activity updates.
01-20 18:31:41.622 22380 22495 D ActivitySM: Registering for activity updates.
01-20 18:31:41.632 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:41.711 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:41.770 22380 22495 I AudioManager: In isSpeakerphoneOn(), calling application: io.homeassistant.companion.android
01-20 18:31:41.771 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=3, streamType=4
01-20 18:31:41.772 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=0
01-20 18:31:41.772 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=1, streamType=3
01-20 18:31:41.773 22380 22495 D AudioManagerExtImpl: getStreamVolume packageName=io.homeassistant.companion.android, index=4, streamType=2
01-20 18:31:41.783 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:41.863 22380 22380 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-20 18:31:41.886 22380 22495 V SystemSensorManagerExtImpl: RegisterListener tcs3408 Ambient Light Sensor Non-wakeup type:5 delay:200000us by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:41.886 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.888 22380 22495 D LightSensor: Light sensor listener registered
01-20 18:31:41.888 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.LightSensorManager
01-20 18:31:41.889 22380 22380 D LightSensor: Light sensor listener unregistered
01-20 18:31:41.889 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.892 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:41.915 22380 22380 D LocBroadcastReceiver: Not requesting accurate location, last accurate location was too recent
01-20 18:31:41.929 22380 22495 D WifiManager: getScanResults check resulttrue
01-20 18:31:41.932 22380 22495 D WifiManager: getScanResults check resulttrue
01-20 18:31:41.936 22380 22495 D NextAlarm: Next alarm is scheduled by net.dinglisch.android.taskerm with trigger time XX
01-20 18:31:41.945 22380 22495 V SystemSensorManagerExtImpl: RegisterListener stk33502 Proximity Sensor Wakeup type:8 delay:200000us by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:41.946 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.951 22380 22495 D ProximitySensor: Proximity sensor listener registered
01-20 18:31:41.952 22380 22495 V SystemSensorManagerExtImpl: RegisterListener pedometer  Non-wakeup type:19 delay:200000us by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:41.952 22380 22495 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.952 22380 22380 V SystemSensorManagerExtImpl: step counter dispatchSensorEvent step 0.0 to + io.homeassistant.companion.android.sensors.StepsSensorManager@6f10403
01-20 18:31:41.953 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.StepsSensorManager
01-20 18:31:41.953 22380 22495 D StepsSensor: Steps sensor listener registered
01-20 18:31:41.953 22380 22380 D StepsSensor: Steps sensor listener unregistered
01-20 18:31:41.953 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:41.956 22380 22495 D StorageSensor: PATHS SD 1
01-20 18:31:41.956 22380 22495 D StorageSensor: PATH /storage/emulated/0/Android/data/io.homeassistant.companion.android/files is mounted true and removable is false
01-20 18:31:41.961 22380 22495 D UrlRepository: localUrl is: true and usesInternalSsid is: true
01-20 18:31:42.040 22380 22380 V SystemSensorManagerExtImpl: unRegisterListener by io.homeassistant.companion.android.sensors.ProximitySensorManager
01-20 18:31:42.041 22380 22380 D ProximitySensor: Proximity sensor listener unregistered
01-20 18:31:42.041 22380 22380 E Parcel  : Reading a NULL string not supported here.
01-20 18:31:42.140 22380 22406 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=341c3df7-5d2c-412c-b79f-429fc7d31d4d, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
01-20 18:31:42.147 22380 22380 I WM-SystemFgDispatcher: Stopping foreground service
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.944 22380 22380 I oplus.android.OplusFrameworkFactoryImpl: get feature:IOplusDynamicVsyncFeature
01-20 18:31:44.952 22380 22380 I Quality : Skipped: false 1
01-20 18:31:44.965 22380 22410 D LogcatReader: Read logcat for pid 22380

```

**Screenshot of problem:**
![Screenshot_2022-01-20-18-22-23-31_49b3d7ede8f829c04ca0dbc08dacb1ac](https://user-images.githubusercontent.com/17238926/150391046-8959bc72-750f-4270-adf5-bd7a7d2aa9db.jpg)

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -163,6 +163,7 @@ dependencies {
     implementation("androidx.navigation:navigation-ui-ktx:2.3.5")
     implementation("com.google.android.material:material:1.4.0")
 
+    implementation("com.squareup.retrofit2:retrofit:2.9.0")
     implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.1")
     implementation("com.squareup.okhttp3:okhttp:4.9.3")
     implementation("com.squareup.picasso:picasso:2.8")

--- a/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
@@ -18,6 +18,8 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.launch
+import retrofit2.HttpException
+import java.util.Calendar
 import java.util.concurrent.Flow
 import java.util.function.Consumer
 import javax.inject.Inject
@@ -40,11 +42,12 @@ class HaControlsProviderService : ControlsProviderService() {
 
     private val domainToHaControl = mapOf(
         "automation" to DefaultSwitchControl,
-        "camera" to null,
         "button" to DefaultButtonControl,
+        "camera" to null,
         "climate" to ClimateControl,
         "cover" to CoverControl,
         "fan" to FanControl,
+        "ha_failed" to HaFailedControl,
         "input_boolean" to DefaultSwitchControl,
         "input_button" to DefaultButtonControl,
         "input_number" to DefaultSliderControl,
@@ -121,7 +124,15 @@ class HaControlsProviderService : ControlsProviderService() {
                                     Log.e(TAG, "Unable to get $it from Home Assistant, null response.")
                                 }
                             } catch (e: Exception) {
-                                Log.e(TAG, "Unable to get $it from Home Assistant, exception.", e)
+                                entities["ha_failed.$it"] = Entity(
+                                    entityId = it,
+                                    state = if (e is HttpException && e.code() == 404) "notfound" else "exception",
+                                    attributes = mapOf<String, String>(),
+                                    lastChanged = Calendar.getInstance(),
+                                    lastUpdated = Calendar.getInstance(),
+                                    context = null
+                                )
+                                Log.e(TAG, "Unable to get $it from Home Assistant, caught exception.", e)
                             }
                         }
                         sendEntitiesToSubscriber(subscriber, entities, areaRegistry, deviceRegistry, entityRegistry)

--- a/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
@@ -113,11 +113,15 @@ class HaControlsProviderService : ControlsProviderService() {
                         var entityRegistry = webSocketRepository.getEntityRegistry()
                         val entities = mutableMapOf<String, Entity<Map<String, Any>>>()
                         controlIds.forEach {
-                            val entity = integrationRepository.getEntity(it)
-                            if (entity != null) {
-                                entities[it] = entity
-                            } else {
-                                Log.e(TAG, "Unable to get $it from Home Assistant.")
+                            try {
+                                val entity = integrationRepository.getEntity(it)
+                                if (entity != null) {
+                                    entities[it] = entity
+                                } else {
+                                    Log.e(TAG, "Unable to get $it from Home Assistant, null response.")
+                                }
+                            } catch (e: Exception) {
+                                Log.e(TAG, "Unable to get $it from Home Assistant, exception.", e)
                             }
                         }
                         sendEntitiesToSubscriber(subscriber, entities, areaRegistry, deviceRegistry, entityRegistry)

--- a/app/src/main/java/io/homeassistant/companion/android/controls/HaFailedControl.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/controls/HaFailedControl.kt
@@ -1,0 +1,46 @@
+package io.homeassistant.companion.android.controls
+
+import android.content.Context
+import android.os.Build
+import android.service.controls.Control
+import android.service.controls.DeviceTypes
+import android.service.controls.actions.ControlAction
+import android.service.controls.templates.StatelessTemplate
+import androidx.annotation.RequiresApi
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.common.data.websocket.impl.entities.AreaRegistryResponse
+
+@RequiresApi(Build.VERSION_CODES.R)
+class HaFailedControl {
+    companion object : HaControl {
+        override fun provideControlFeatures(
+            context: Context,
+            control: Control.StatefulBuilder,
+            entity: Entity<Map<String, Any>>,
+            area: AreaRegistryResponse?
+        ): Control.StatefulBuilder {
+            control.setStatus(if (entity.state == "notfound") Control.STATUS_NOT_FOUND else Control.STATUS_ERROR)
+            control.setStatusText("")
+            control.setControlTemplate(
+                StatelessTemplate(
+                    entity.entityId
+                )
+            )
+            return control
+        }
+
+        override fun getDeviceType(entity: Entity<Map<String, Any>>): Int =
+            DeviceTypes.TYPE_UNKNOWN
+
+        override fun getDomainString(context: Context, entity: Entity<Map<String, Any>>): String =
+            entity.entityId.split(".")[0].capitalize()
+
+        override fun performAction(
+            integrationRepository: IntegrationRepository,
+            action: ControlAction
+        ): Boolean {
+            return false
+        }
+    }
+}

--- a/app/src/main/java/io/homeassistant/companion/android/qs/TileExtensions.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/qs/TileExtensions.kt
@@ -74,8 +74,13 @@ abstract class TileExtensions : TileService() {
                         tile.subtitle = tileData.subtitle
                     }
                     if (tileData.entityId.split('.')[0] in toggleDomains) {
-                        val state = runBlocking { integrationUseCase.getEntity(tileData.entityId) }
-                        tile.state = if (state?.state == "on" || state?.state == "open") Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
+                        try {
+                            val state = runBlocking { integrationUseCase.getEntity(tileData.entityId) }
+                            tile.state = if (state?.state == "on" || state?.state == "open") Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
+                        } catch (e: Exception) {
+                            Log.e(TAG, "Unable to set state for tile", e)
+                            tile.state = Tile.STATE_UNAVAILABLE
+                        }
                     } else
                         tile.state = Tile.STATE_INACTIVE
                     val iconId = tileData.iconId

--- a/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt
@@ -173,15 +173,15 @@ class CameraWidget : AppWidgetProvider() {
     }
 
     private suspend fun retrieveCameraImageUrl(context: Context, entityId: String): String? {
-        val entity = integrationUseCase.getEntity(entityId)
-        if (entity == null) {
+        return try {
+            val entity = integrationUseCase.getEntity(entityId)
+            entity?.attributes?.get("entity_picture")?.toString()
+        } catch (e: Exception) {
             Log.e(TAG, "Failed to fetch entity or entity does not exist")
             if (lastIntent == UPDATE_IMAGE)
                 Toast.makeText(context, commonR.string.widget_entity_fetch_error, Toast.LENGTH_LONG).show()
-            return null
+            null
         }
-
-        return entity.attributes["entity_picture"]?.toString()
     }
 
     override fun onReceive(context: Context, intent: Intent) {

--- a/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt
@@ -177,7 +177,7 @@ class CameraWidget : AppWidgetProvider() {
             val entity = integrationUseCase.getEntity(entityId)
             entity?.attributes?.get("entity_picture")?.toString()
         } catch (e: Exception) {
-            Log.e(TAG, "Failed to fetch entity or entity does not exist")
+            Log.e(TAG, "Failed to fetch entity or entity does not exist", e)
             if (lastIntent == UPDATE_IMAGE)
                 Toast.makeText(context, commonR.string.widget_entity_fetch_error, Toast.LENGTH_LONG).show()
             null

